### PR TITLE
Added initial test case for User/Group management with role-mappings

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:
This PR contains a sample test case which is essential for User and Group management using keycloak APIs. Also, we will be assigned Px-Backup specific roles to groups and validating it.

Changes include:

- Sample test case for keycloak user/group management and role-mappings
- Includes environment variables for backup api and keycloak endpoint
- Logic to utilise backup and keycloak endpoint when running locally or from a VM outside of the k8s cluster
- Added functions like GetRolesForUser, ValidateUserRole, AddRoleToGroup, DeleteRoleFromGroup, DeleteUser, DeleteGroup
- Formatting a few comments

[Testrun link](http://aetos.pwx.purestorage.com/resultSet/testSetID/21167) 

**Which issue(s) this PR fixes** (optional)
Closes # PA-308

**Special notes for your reviewer**:

- The sample test case is just a building block for the real system tests that will be added in the coming days. These functions will be used mainly in the Share Backup feature where we need to test with multiple users and groups with different levels of access to the shared backup
- Please go easy since this is my first PR 😄

### CODE REVIEW

File: drivers/backup/auth.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/sirupsen/logrus"
```
Comment (lsrinivas-pure) [inline drivers/backup/auth.go:7]: rebase with latest changes and use pkg/log

File: drivers/backup/auth.go [hunk 2]
```diff
@@ -264,8 +282,68 @@ func GetAllRoles() ([]KeycloakRoleRepresentation, error) {
...
+		return nil, err
+	}
+	return roles, nil
+}
+
+// ValidateUserRole will validate if a given user has the provided PxBackupRole mapped to it
+func ValidateUserRole(userName string, role PxBackupRole) bool {
+	roleMapping, err := GetRolesForUser(userName)
+	if err != nil {
+		return false
```
Comment (lsrinivas-pure) [inline drivers/backup/auth.go:321]: retruning just false we will miss to log the error, fail it here logging the error or return the error

File: drivers/backup/auth.go [hunk 3]
```diff
@@ -264,8 +282,68 @@ func GetAllRoles() ([]KeycloakRoleRepresentation, error) {
...
+	err = json.Unmarshal(response, &roles)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return nil, err
+	}
+	return roles, nil
+}
+
+// ValidateUserRole will validate if a given user has the provided PxBackupRole mapped to it
+func ValidateUserRole(userName string, role PxBackupRole) bool {
```
Comment (lsrinivas-pure) [inline drivers/backup/auth.go:318]: Validate can we move it to some common files , rather than in drivers

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -71,7 +70,72 @@ var _ = Describe("{BackupClusterVerification}", func() {
...
+// This is a sample test case to verify User/Group Management and role mapping
+var _ = Describe("{UserGroupManagement}", func() {
+	JustBeforeEach(func() {
+		log.Infof("No pre-setup required for this testcase")
+		StartTorpedoTest("Backup: UserGroupManagement", "Creating users and adding them to groups", nil, 0)
+	})
+	It("User and group role mappings", func() {
+		Step("Create Users", func() {
+			err := backup.AddUser("testuser1", "test", "user1", "testuser1@localhost.com", "Password1")
+			if err != nil {
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:82]: use log.FailOnError ,so that we can avoid err !=nil check

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: drivers/backup/auth.go (hunk 1 above)
Comment (mkoppal-px) [reply drivers/backup/auth.go]: Addressed

File: drivers/backup/auth.go (hunk 3 above)
Comment (mkoppal-px) [reply drivers/backup/auth.go]: Done. Moved it to common.go

File: drivers/backup/auth.go (hunk 2 above)
Comment (mkoppal-px) [reply drivers/backup/auth.go]: Now returning bool and err. Then handling error and failing it.

File: tests/backup/backup_test.go (hunk 4 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go]: Done. Makes sense.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — COMMENTED** [review]
A few things we can do is:

- [ ] Add group to group mapping function which cane be used. 
- [ ] Add group to default role mapping function. 
- [ ] Add user to default role mapping function.

This will be helpful for us to test scenarios like OIDC to keycloak mapping scenarios.

File: drivers/backup/auth.go [hunk 5]
```diff
@@ -54,6 +54,8 @@ const (
 	pxBackupNamespace = "PX_BACKUP_NAMESPACE"
 	// OidcSecretName where secrets for OIDC auth cred info resides
 	oidcSecretName = "SECRET_NAME"
+	// PxCentralUI URL Eg: http://10.13.114.233:32482
```
[the remark this answers is not in the document]
Comment (mkoppal-px) [reply drivers/backup/auth.go]: Done

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Comment by mkoppal-px** [comment]
> A few things we can do is:
> 
> * [ ]  Add group to group mapping function which cane be used.
> * [ ]  Add group to default role mapping function.
> * [ ]  Add user to default role mapping function.
> 
> This will be helpful for us to test scenarios like OIDC to keycloak mapping scenarios.

@kshithijiyer-px Created PA tickets for these - PA-404 and PA-405

**Pull request merged**